### PR TITLE
Fixes #11314: Node compliance are never archived

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/11314

### CODE REVIEW

File: rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ExpectedReportsJdbcRepository.scala [hunk 1]
```diff
@@ -400,6 +400,70 @@ class UpdateExpectedReportsJdbcRepository(
...
+     }
+  }
+
+  /**
+   * Delete all node compliance whose run timestamp is before a date
+   */
+  override def deleteNodeCompliances(date: DateTime) : Box[Int] = {
+
+    val dateAt_0000 = date.toString("yyyy-MM-dd")
+    val d1 = s"delete from archivednodecompliance where coalesce(runtimestamp, '${dateAt_0000}') < '${dateAt_0000}'"
```
Comment (ncharles) [inline rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ExpectedReportsJdbcRepository.scala]: i think you meant nodecompliance

**Comment by fanf** [comment]
PR rebased

File: rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ExpectedReportsJdbcRepository.scala [hunk 2]
```diff
@@ -400,6 +400,70 @@ class UpdateExpectedReportsJdbcRepository(
...
+    val copy = s"""
+      insert into archivednodecompliance
+        (nodeid, runtimestamp, endoflife, runanalysis, summary, details)
+        (select nodeid, runtimestamp, endoflife, runanalysis, summary, details
+           from nodecompliance
+           where coalesce(runtimestamp, '${dateAt_0000}') < '${dateAt_0000}'
+        )
+        """
+    val delete = s"""
+      delete from archivednodecompliance where coalesce(runtimestamp, '${dateAt_0000}') < '${dateAt_0000}'
```
Comment (ncharles) [inline rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ExpectedReportsJdbcRepository.scala]: it should be nodecompliance, but I guess you'd save a lot of disk space :)

**Comment by fanf** [comment]
PR rebased

**Pull request merged**